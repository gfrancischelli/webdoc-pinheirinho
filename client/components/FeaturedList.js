import React from 'react';


// const FeaturedList = (props) => (
//   <div>
//     birbi
//   </div>
// )


class FeaturedList extends React.Component {
  render() {
    return (
<div>
    birbi
  </div>
    )
  }
}

export default FeaturedList;
