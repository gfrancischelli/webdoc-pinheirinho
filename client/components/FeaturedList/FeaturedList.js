import React from 'react';
import { Link } from 'react-router';
import ImageSet from 'components/ImageSet/ImageSet';

import {imageURL, resourceURL} from 'utils';

function FeaturedList({ featured, type }) {
  return (
    <ul className='o-feat-list'>
      { featured.map( (item, index) => (
        <li key={ index } className='o-feat-list__item o-card'>
          { !item.heroImage ? null :
            <ImageSet 
              url={ imageURL(type, item.heroImage.filename)}
              className='o-card__figure'
            />
          }
          <div className='o-card__body'>
            <Link to={ resourceURL(type, item.slug) }>
              <h3 className='c-heading c-heading--small'>
                { item.title }
              </h3>
            </Link>
            { !item.content ? null :
              <p className='summary' 
                dangerouslySetInnerHTML={
                  {__html: item.content.slice(0, 220)}
                } />
            }
          </div>
        </li>
      ))}
    </ul>
  )  
}

FeaturedList.propTypes = {
  featured: React.PropTypes.array
}

export default FeaturedList;
