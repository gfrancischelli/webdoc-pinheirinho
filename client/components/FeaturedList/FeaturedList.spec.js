import React from 'react';
import FeaturedList from './FeaturedList';

const mock_data =[{
    heroUrl: 'dummy hero',
    title: 'dummy title',
    content: 'dummy content'
  },{
    heroUrl: 'dummy hero',
    title: 'dummy title',
    content: 'dummy content'
  },{
    heroUrl: 'dummy hero',
    title: 'dummy title',
    content: 'dummy content'
}];

it('FeaturedList renders correctly', () => {
  const wrapper = shallow(
    <FeaturedList featured={mock_data} />
  )
  expect(wrapper).toMatchSnapshot();
});
