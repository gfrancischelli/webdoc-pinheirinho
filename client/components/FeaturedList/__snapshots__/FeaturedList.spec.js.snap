exports[`test FeaturedList renders correctly 1`] = `
<ul
  className="o-feat-list">
  <li
    className="o-feat-list__item o-card">
    <div
      className="o-card__body">
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/noticias">
        <h3
          className="c-heading c-heading--small" />
      </Link>
      <p
        className="summary"
        dangerouslySetInnerHTML={
          Object {
            "__html": "dummy content",
          }
        } />
    </div>
  </li>
  <li
    className="o-feat-list__item o-card">
    <div
      className="o-card__body">
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/noticias">
        <h3
          className="c-heading c-heading--small" />
      </Link>
      <p
        className="summary"
        dangerouslySetInnerHTML={
          Object {
            "__html": "dummy content",
          }
        } />
    </div>
  </li>
  <li
    className="o-feat-list__item o-card">
    <div
      className="o-card__body">
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/noticias">
        <h3
          className="c-heading c-heading--small" />
      </Link>
      <p
        className="summary"
        dangerouslySetInnerHTML={
          Object {
            "__html": "dummy content",
          }
        } />
    </div>
  </li>
</ul>
`;
